### DISCUSSION TITLE
ModuleNotFoundError

### DISCUSSION THREAD

**Discussion by colingoodman** [pytest-dev/pytest#10105] [Q&A] [answered]
At some point in the last few months pytest broke in an existing repo of mine. I am currently using `Python 3.9`, `Pytest 7.1.2`. Running `pytest` or `python -m pytest` yields 

```
...
/path/project/test/test_main.py:1: in <module>
    ???
../../venv/lib/python3.9/site-packages/fastapi/testclient.py:1: in <module>
    from starlette.testclient import TestClient as TestClient  # noqa
../../venv/lib/python3.9/site-packages/starlette/testclient.py:16: in <module>
    import requests
E   ModuleNotFoundError: No module named 'requests'
```

I am using FastAPI (accordingly basing my tests originally off [this](https://fastapi.tiangolo.com/tutorial/testing/)) among a few other libraries in this project. All of them are installed properly in my venv; I can run the project with no issues.

I found [this](https://medium.com/@dirk.avery/pytest-modulenotfounderror-no-module-named-requests-a770e6926ac5) suggestion online of removing pytest from my global pip packages, but it did not resolve the problem. I have also tried adding and removing `__init__.py` files in the tests directory and the main project directory with no success

Thank you for for any help/suggestions.

**Comment by RonnyPfannschmidt** [comment, marked as answer]
What does importing request standalone do? It seems like there is a problem being hidden by the error handling

**Reply by colingoodman** [reply]
Your suggestion lead me [here](https://github.com/tiangolo/fastapi/issues/2595) where I found out that the Starlette testclient class requires requests, but this oddly doesn't get installed as a dependency by anything. Installing requests fixed this. Thanks!

**Reply by RonnyPfannschmidt** [reply]
Thanks for the feedback, marking this as answer

Happy testing
